src/common/Notifications.ts

```diff
@@ -0,0 +1,107 @@
+import { AxiosInstance } from 'axios';
+import { Platform } from 'react-native';
+import {
+  Notifications,
+  Registered,
+  Notification,
+  NotificationCompletion,
+  RegistrationError,
+  NotificationBackgroundFetchResult,
+} from 'react-native-notifications';
+
+export const registerDeviceToken = ({
+  deviceToken,
+  application,
+  httpClient,
+}: {
+  deviceToken: string;
+  application: string;
+  httpClient: AxiosInstance;
+}) => {
+  const provider = Platform.OS === 'ios' ? 'APNS' : 'GCM';
+  const params = {
+    provider,
+    deviceToken,
+    application,
+  };
+  httpClient.post('/device-endpoints', params);
+};
+
+export const getInitialNotification = () => {
+  return Notifications.getInitialNotification();
+};
+
+export const isRegisteredForNotifications = () => {
+  return Notifications.isRegisteredForRemoteNotifications();
+};
+
+export const requestNotificationsPermissions = (
+  callback: ({
+    deviceToken,
+    denied,
+    error,
+  }: {
+    deviceToken?: string;
+    denied?: boolean;
+    error?: RegistrationError;
+  }) => void,
+) => {
```

Comment:
Did you consider using a Promise ctor instead of callback?

Reply:
I did. I went with callbacks because they're the closest to the two most popular RN push notifications libraries APIs, [react-native-notifications](https://wix.github.io/react-native-notifications/api/general-events#registerremotenotificationsregistered) and [notifee](https://notifee.app/react-native/docs/events).